File: UWSN/Model/Frame.cs
﻿using System.Numerics;

namespace UWSN.Model;

public class Frame
{
    public const double FRAME_SIZE_IN_BITS = 256.0;

    public enum FrameType
    {
        Ack,
        Data,
        Hello,
        Warning
    }

    public required FrameType Type { get; set; }

    public required int SenderId { get; set; }

    public required Vector3 SenderPosition { get; set; }

    public required int ReceiverId { get; set; }

    public required DateTime TimeSend { get; set; }

    public required bool AckIsNeeded { get; set; }

    public required double BatteryLeft { get; set; }

    public required CollectedData? CollectedData { get; set; }

    public required Dictionary<int, Sensor.Neighbour>? NeighboursData { get; set; }

    public required List<int>? DeadSensors { get; set; }
}

File: client/src/renderer/shared/types/frame.ts
import { CollectedData } from './collectedData'
import { Vector3 } from './vector3'

export enum FrameType {
    RegularFrame = 'RegularFrame',
    Ack = 'Ack',
    Data = 'Data',
    Hello = 'Hello',
    Warning = 'Warning'
}

export type Frame = {
    Type: FrameType
    SenderId: number
    SenderPosition: Vector3
    ReceiverId: number
    TimeSend: string
    AckIsNeeded: boolean
    CollectedData: CollectedData | undefined
    //NeighboursData: Neighbour[] | undefined
    //DeadSensors: number[] | undefined
}
